File: python/button.py
from time import sleep
import RPi.GPIO as GPIO #gives us access to the GPIO pins
GPIO.setmode(GPIO.BCM)  #sets the pin numbering system we want to use

button = 25  #set variable for the pin number used

GPIO.setup(button, GPIO.IN)  #configure GPIO pin to input mode

while True:
    if GPIO.input(button):
        print("Button pressed")
    else:
        print("Button not pressed")
    sleep(0.1)

File: wiringPi/button/button.c
#include <wiringPi.h>
#include <stdio.h>

const int button = 6;

void setup()
{
	pinMode(button, INPUT);
}

int main(void)
{
	if(wiringPiSetup() == -1)
		return 1;
	
	while(1)
	{
		if (digitalRead(button))
            printf("button pressed");

		else
            printf("button not pressed");

		delay(100);
	}
	return 0;
}
